### PR TITLE
Bugfix code logic to allow for passing of `nn.Module` to  `TorchFXFeatureExtractor`

### LINKED ISSUE #933: [Bug]: TorchFXFeatureExtractor Does not handle `nn.Module` passed to __init__ properly
### Describe the bug

According to typing, `TorchFXFeatureExtractor`should accept `backbone: str | BackboneParams | dict | nn.Module,`. However, when passing nn.Module to it, it fails.

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

```
import timm
from anomalib.models.components.feature_extractors import TorchFXFeatureExtractor

model = timm.create_model("resnet18", pretrained=True)
TorchFXFeatureExtractor(model, ["layer1"]) # is expected to pass
```

### OS information

OS information:
- OS: [e.g. Ubuntu 20.04]
- Python version: [e.g. 3.8.10]
- Anomalib version: [e.g. 0.3.6]
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: [e.g. 2x GeForce RTX 3090]
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

Valid featureextractor is generated

### Screenshots

_No response_

### Pip/GitHub

pip

### What version/branch did you use?

_No response_

### Configuration YAML

```yaml
N/A
```

### Logs

```shell
N/A
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ORippler**
# Description

- Provide a summary of the modification as well as the issue that has been resolved. List any dependencies that this modification necessitates.
Fix code logic to allow for passing of `nn.Module` to  `TorchFXFeatureExtractor`

- Fixes #933

## Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

File: anomalib/models/components/feature_extractors/torchfx.py [hunk 1]
```diff
@@ -79,14 +95,18 @@ def __init__(
         super().__init__()
         if isinstance(backbone, dict):
             backbone = BackboneParams(**backbone)
-        elif not isinstance(backbone, BackboneParams):  # if str or nn.Module
+        elif isinstance(backbone, str):
             backbone = BackboneParams(class_path=backbone)
+        elif not isinstance(backbone, nn.Module) and not isinstance(backbone, BackboneParams):
```
Comment (samet-akcay) [inline anomalib/models/components/feature_extractors/torchfx.py:100]:
```suggestion
        elif not isinstance(backbone, (nn.Module, BackboneParams)):
```

**Review by samet-akcay — CHANGES_REQUESTED** [review]
Thanks @ORippler! I have a single comment only. I'll let @ashwinvaidya17 review the PR.

**Review by ashwinvaidya17 — CHANGES_REQUESTED** [review]
Thanks for spotting and addressing this! Can you add the example to the tests as well https://github.com/openvinotoolkit/anomalib/blob/main/tests/pre_merge/models/test_feature_extractor.py#L47. Just add it at the end. I'll refactor that test later on and split each torchfx tests.

**Comment by ORippler** [comment]
I would suggest changing `BackboneParams.class_path` in another Pull Request to be more descriptive (it can be both a path to a class and a nn.Module class-object)

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks. I am fine with the changes.

**Review by samet-akcay — APPROVED** [review]
Thanks again for your contribution!

**Pull request merged**